### PR TITLE
Align spring-security-pac4j with pac4j version 1.9.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by mirceac**
spring-security-pac4j is done using pac4j 1.8.8-SNAPSHOT version

This pull request aligns spring-security-pac4j with pac4j 1.9.0-SNAPSHOT version (latest)

Please have a look - thanks

### CODE REVIEW

**Comment by leleuj** [comment]
Thanks for your help. My plan is to first release pac4j v1.8.8 and spring-security-pac4j v1.4.3. Then, I'll create a 1.4.x branch and update the master in version 1.5.0-SNAPSHOT.
The upgrade should have a broader scope than your proposal: I will make a few comments and we'll see how far you can go before I take over.

File: pom.xml [hunk 1]
```diff
@@ -57,7 +57,7 @@
         <spring.security.version>4.0.3.RELEASE</spring.security.version>
```
Comment (leleuj) [inline pom.xml]: You can update all dependencies to their latest versions and especially to Java 8.

You must take a look at the pac4j pom.xml file and copy the pmd and findbugs plugin.

File: src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProvider.java [hunk 2]
```diff
@@ -75,7 +77,12 @@ public Authentication authenticate(final Authentication authentication) throws A
         final Client client = this.clients.findClient(clientName);
         // get the user profile
         final WebContext context = token.getContext();
-        final UserProfile userProfile = client.getUserProfile(credentials, context);
+        UserProfile userProfile = null;
+        try {
+            userProfile = client.getUserProfile(credentials, context);
+        } catch (RequiresHttpAction requiresHttpAction) {
+            throw new AuthenticationCredentialsException(requiresHttpAction.getMessage(), requiresHttpAction);
```
Comment (leleuj) [inline src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProvider.java]: In fact, no. We need to treat `RequiresHttpAction` as real HTTP actions. Maybe `return;` should be enough. For sure, it should not generate an exception. Maybe add a `TODO` for later update.

File: src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationToken.java [hunk 3]
```diff
@@ -97,10 +97,10 @@ public Object getPrincipal() {
 
     @Override
     public void eraseCredentials() {
-        credentials.clear();
-        if (userProfile != null) {
-            userProfile.clear();
-        }
+        //credentials.clear();
```
Comment (leleuj) [inline src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationToken.java]: `credentials = null;` and then, if user profile is not, call the `clearSensitiveData` method.

File: src/test/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProviderTest.java [hunk 4]
```diff
@@ -27,6 +27,7 @@
 import org.pac4j.core.client.Client;
```
Comment (leleuj) [inline src/test/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProviderTest.java]: I'm not fully convinced by all tests. Maybe I'll rewrite them later on.

File: pom.xml (hunk 1 above)
Comment (mirceac) [inline pom.xml]: I will update with spring security 4.0.4.RELEASE version and spring to
4.2.5.RELEASE. these are the latest in maven repository (released february 2016)

http://mvnrepository.com/artifact/org.springframework.security/spring-security-core
http://mvnrepository.com/artifact/org.springframework/spring-context

File: src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProvider.java (hunk 2 above)
Comment (mirceac) [inline src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProvider.java]: I will log a debug message and return null instead

File: src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationToken.java (hunk 3 above)
Comment (mirceac) [inline src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationToken.java]: will do

File: src/test/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProviderTest.java (hunk 4 above)
Comment (mirceac) [inline src/test/java/org/pac4j/springframework/security/authentication/ClientAuthenticationProviderTest.java]: ok - thanks

**Comment by leleuj** [comment]
Be careful, I just committed the update to use pac4j v1.8.8 (I just released it), you may need to sync with master...

**Comment by leleuj** [comment]
and fix conflicts :-(

**Comment by mirceac** [comment]
sure, no problem

**Comment by leleuj** [comment]
@garpinc @mirceac  I just released version 1.4.3 (based on pac4j v1.8.8) and upgraded the master to version 1.5.0-SNAPSHOT.

**Comment by mirceac** [comment]
synchronized with latest && commited required changes... please have a look

File: pom.xml [hunk 5]
```diff
@@ -54,10 +54,10 @@
     </developers>
 
     <properties>
-        <spring.security.version>4.0.3.RELEASE</spring.security.version>
-        <spring.version>4.2.2.RELEASE</spring.version>
+        <spring.security.version>4.0.4.RELEASE</spring.security.version>
+        <spring.version>4.2.5.RELEASE</spring.version>
         <servlet.version>3.0.1</servlet.version>
```
Comment (leleuj) [inline pom.xml]: We could upgrade to servlet 3.1, don't you think?

**Comment by leleuj** [comment]
You also need to update the .travis.yml file with oraclejdk8 otherwise the build won't work.

**Comment by mirceac** [comment]
Done. upgraded servlet to 3.1.0 and oraclejdk8. build passes now

**Comment by leleuj** [comment]
Great! I will merge the pull request "as is".

We'll have a few remaining topics to address later on: multi profiles support, better naming of the components...

Can you upgrade the demo as well: https://github.com/pac4j/spring-security-pac4j-demo? Thanks

**Comment by mirceac** [comment]
thanks for accepting contribution! I will come back with pull request for the spring-security-pac4j-demo

**Comment by leleuj** [comment]
I you feel interested, I can drive you on the remaining issues we need to address...

**Pull request merged**